### PR TITLE
Allow existing secret

### PR DESCRIPTION
**Pull request by davidholsgrove**
* Allow use of existing secret - defined outside of the helm values.yaml
* Provide example renovate.config value
* Use _helper defined serviceaccountname
* Use nindent to improve readability and the existing _helper defined labels

### CODE REVIEW

**Comment by davidholsgrove** [comment]
Not sure why kubeval test fails - recreating locally;

```
helm template --values "${CHART_DIR}"/ci/ci-values.yaml "${CHART_DIR}" | kubeval --strict --ignore-missing-schemas --kubernetes-version 1.14.10 --schema-location "${SCH
EMA_LOCATION}"
WARN - Set to ignore missing schemas
PASS - renovate/templates/configmap.yaml contains a valid ConfigMap (RELEASE-NAME-renovate-config)
PASS - renovate/templates/cronjob.yaml contains a valid CronJob (RELEASE-NAME-renovate)
```

File: charts/renovate/values.yaml [hunk 1]
```diff
@@ -19,8 +19,33 @@ image:
 
 renovate:
   config: ''
+  # See https://docs.renovatebot.com/self-hosted-configuration
+  # config: |
+  #   {
+  #     "platform": "gitlab",
+  #     "endpoint": "https://gitlab.example.com/api/v4",
+  #     "timezone": "Australia/Brisbane",
```
Comment (MindTooth) [inline charts/renovate/values.yaml:27]: Is `timezone` a self-hosted valid option?

Can't find it [here](https://docs.renovatebot.com/self-hosted-configuration/).  Just curious. 😄

Comment (davidholsgrove) [reply charts/renovate/values.yaml:27]: Ah, no you're right - its a https://docs.renovatebot.com/configuration-options/#timezone option, removed thanks :)

Comment (JamieMagee) [reply charts/renovate/values.yaml]: Most configuration options are also valid self-hosted configuration options :)

**Comment by JamieMagee** [comment]
@davidholsgrove Thanks for the contribution. This all looks good to me. Let's figure out why kubeval is failing, and then we can get this merged.

**Comment by JamieMagee** [comment]
I added `set -x` to the shell scripts. Hopefully this gives some more insight into the failure.

**Comment by JamieMagee** [comment]
Okay, so it looks like the changed chart is failing in being picked up in the command `"$(git diff --find-renames --name-only "$(git rev-parse --abbrev-ref HEAD)" remotes/origin/master -- charts | grep '[cC]hart.yaml' | sed -e 's#/[Cc]hart.yaml##g')"`, and due to `set -o pipefail` the whole script is returning with `exit 1`.

Thing is, that also returns correctly on my machine

```sh
$ echo "$(git diff --find-renames --name-only "$(git rev-parse --abbrev-ref HEAD)" remotes/origin/master -- charts | grep '[cC]hart.yaml' | sed -e 's#/[Cc]hart.yaml##g')"
charts/renovate
```

Curious 🤔

**Comment by JamieMagee** [comment]
Okay, I've changed that check for something that seems to be a bit more reliable. It will also check for any changes in the `charts` directory, instead of looking for changes in `Chart.yaml` specifically.

File: charts/renovate/values.yaml [hunk 2]
```diff
@@ -19,8 +19,32 @@ image:
...
+  #     "logLevel": "debug",
+  #     "repositories": ["username/repo", "orgname/repo"]
+  #   }
 
+# Provide secrets inline
 secrets: {}
+# or provide the name of an existing secret to be read instead.
+existingSecret: ""
+
+serviceAccount:
```
Comment (MindTooth) [inline charts/renovate/values.yaml:40]: The following addition too broke my working deployment via GitLab managed cluster.  Tried reading up on the best practice, but could not find any.  The temporary (permanent) solution were to set `create: false` instead.  GitLab manages this already.

I could make a PR with additional notes on how this affects e.g. GitLab managed cluster, etc.?

Comment (viceice) [reply charts/renovate/values.yaml:40]: Feel free to open a PR to fix the values.yaml. I think the service account can be completely optional.

Comment (JamieMagee) [reply charts/renovate/values.yaml:40]: @MindTooth Can you share any logs?

Comment (MindTooth) [reply charts/renovate/values.yaml:40]: The only error that I get anywhere:
> Error creating: pods "renovate-bot-manual-hg6-" is forbidden: error looking up service account renovate-bot-68-production/renovate-bot: serviceaccount "renovate-bot" not found

`renovate-bot-68-production-service-account` is created by GitLab and should be used.  Hence it don't have permission to continue.

Comment (JamieMagee) [reply charts/renovate/values.yaml:40]: Okay, I looked, and I'd agree with you. `create: false` looks like the right fix here. The question is, should we set this as `false` by default, or leave it as `true` and add something to the documentation?

any input @davidholsgrove?

Comment (davidholsgrove) [reply charts/renovate/values.yaml:40]: Ah yes I see whats happened - the existing reference to `renovate.serviceAccountName` in the `_helpers.tpl` points to a template that doesnt exist. I've opened https://github.com/renovatebot/helm-charts/issues/65 to add the missing template

**Pull request merged**